sympy/integrals/integrals.py

```diff
@@ -516,6 +517,44 @@ def try_meijerg(function, xab):
                             function = ret
                             continue
 
+            if not isinstance(antideriv,Integral) and antideriv is not None:
+                sym = xab[0]
+                if antideriv.is_Add:
+                    terms = antideriv.args
+                else:
+                    terms = tuple((antideriv,))
+                for term in terms:
+                    atan_part = term.atoms(atan)
+                    if atan_part and len(atan_part)==1:
+                        atan_arg = atan_part.pop().args[0]
+                        # Checking `atan_arg` to be linear combination of `tan` or `cot`
+                        if atan_arg.has(tan):
+                            x1 = Dummy('x1')
+                            tan_part = atan_arg.atoms(tan).pop()
+                            tan_exp1 = atan_arg.subs(tan_part,x1)
+                            # If `tan_part` has any other function then `tan`,
+                            # then diff() raises `ValueError`
+                            try:
+                                tan_exp1.diff()
```

Comment:
This should be `tan_exp1.diff(x1)` to avoid confusion. It should probably also be checked that the derivative is constant.

Reply:
Case when `atan` has more than one function of `tan(x)`:
```
>>> exp = atan(sqrt(2)*(tan(x/2)+sqrt(2)*x))
>>> exp
atan(sqrt(2)*(sqrt(2)*x + tan(x/2)))
>>> arg = exp.atoms(atan).pop().args[0]
>>> arg
sqrt(2)*(sqrt(2)*x + tan(x/2))
>>> arg = arg.subs(arg.atoms(tan).pop(),x1)
>>> arg
sqrt(2)*(_x1 + sqrt(2)*x)
>>> arg.diff()    # This gives ValueError because more than one variable is present
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/core/expr.py", line 3008, in diff
    return Derivative(self, *new_symbols, **assumptions)
  File "sympy/core/function.py", line 1125, in __new__
    must be supplied to differentiate %s''' % expr))
ValueError: 
Since there is more than one variable in the expression, the
variable(s) of differentiation must be supplied to differentiate
sqrt(2)*(_x1 + sqrt(2)*x)
>>> arg
sqrt(2)*(_x1 + sqrt(2)*x)
>>> arg.diff(x1)    # If we differentiate w.r.t. `x1` then it wont give the desired answer
sqrt(2)
```

Case when `atan` is linear function of `tan(x)`:
```
>>> exp = atan(sqrt(2)*(tan(x/2)+sqrt(2)))
>>> exp
atan(sqrt(2)*(sqrt(2) + tan(x/2)))
>>> arg = exp.atoms(atan).pop().args[0]
>>> arg
sqrt(2)*(sqrt(2) + tan(x/2))
>>> arg = arg.subs(arg.atoms(tan).pop(),x1)
>>> arg
sqrt(2)*(_x1 + sqrt(2))
>>> arg.diff()
sqrt(2)
```

Reply:
> If we differentiate w.r.t. `x1` then it wont give the desired answer

Why would `sqrt(2)` not be the desired answer?

Reply:
If arg is non-linear function of `tan` (ex- `arg = sqrt(2)*(_x1 + sqrt(2)*x)` ), then it's differentiation treat other symbols as constant, and gives `sqrt(2)`.
If `arg` will be linear function in `tan` then it while substitution with dummy variable. The arg will only be a function of `x1`(dummy), and if not a linear function in `tan` then it will give `ValueError` because of more than one variable

Reply:
I think that the most important matter is that the coefficient of `x1` be constant. The rest (namely `arg.subs(x1, 0)`) need not necessarily be constant if it is non-singular. The singular values `+/-oo` of `arg` come then from `x1` alone. The sign of the derivative is important since it will determine whether `arg` goes from `-oo` to `oo` or from `oo` to `-oo`. In the latter case the sign of the floor term has to be changed.

Reply:
ok!!